File: everyshowing/src/main/java/com/wghcwc/everyshowing/activitystack/ActivityChangeListener.java
package com.wghcwc.everyshowing.activitystack;

import android.app.Activity;

/**
 * @author wghcwc
 * @date 19-12-5
 */
public interface ActivityChangeListener {


    /**
     * 当前Activity Destroy
     * @return true 删除当前listener
     * false 不变
     *
     * @param activity 当前Activity
     *
     */
    void onActivityDestroy(Activity activity) ;

    /**
     * 当前Activity 创建
     *
     * @param activity
     */
    void onActivityCreate(Activity activity);

    void onActivitySateChange(Activity activity, ActivityState state);

}

File: activitylifecycle2/src/main/java/com/wghcwc/activitylifecycle2/ActivityChangeListener.kt
package com.wghcwc.activitylifecycle2

import android.app.Activity

/**
 * @author wghcwc
 * @date 19-12-5
 */
interface ActivityChangeListener {
    /**
     * 当前Activity Destroy
     *
     * @param activity 当前Activity
     * @return true 删除当前listener
     * false 不变
     */
    fun onActivityDestroy(activity: Activity)

    /**
     * 状态改变
     *
     * @param activity
     * @param state
     */
    fun onActivitySateChange(activity: Activity, state: ActivityState)
}